### PR TITLE
[SPARK-47307][DOCS][FOLLOWUP] Add a migration guide for the behavior change of base64 function

### PR DESCRIPTION
**Pull request by wForget**
### What changes were proposed in this pull request?

Follow up to #47303

Add a migration guide for the behavior change of `base64` function

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

doc change

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

**Comment by wForget** [comment]
cc @yaooqinn

File: docs/sql-migration-guide.md [hunk 1]
```diff
@@ -64,6 +64,7 @@ license: |
 ## Upgrading from Spark SQL 3.5.1 to 3.5.2
 
 - Since 3.5.2, MySQL JDBC datasource will read TINYINT UNSIGNED as ShortType, while in 3.5.1, it was wrongly read as ByteType.
+- Since 3.5.2, the `base64` function will return a non-chunked string. To restore the previous behavior where base64 encoded strings that chunked into lines of at most 76 characters, set `spark.sql.legacy.chunkBase64String.enabled` to `true`.
```
Comment (pan3793) [inline docs/sql-migration-guide.md:67]: the behavior changed in spark 3.3.0, so "the previous behavior" might be ambiguous.

Comment (wForget) [reply docs/sql-migration-guide.md]: Makes sense, I made a little change, thank you.

**Review by yaooqinn — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
@allisonwang-db wanna try merging this?

**Comment by allisonwang-db** [comment]
Merged to master and branch-3.5

**Pull request closed without merging**